From: Will Deacon <will@kernel.org>
Date: Fri, 4 Oct 2019 14:08:13 +0100
Subject: arm64: Default to building compat vDSO with clang when
 CONFIG_CC_IS_CLANG
Git-commit: 24ee01a927bfe56c66429ec4b1df6955a814adc8
Patch-mainline: 5.4-rc3
References: bnc#1151927 5.3.9

Rather than force the use of GCC for the compat cross-compiler, instead
extract the target from CROSS_COMPILE_COMPAT and pass it to clang if the
main compiler is clang.

Acked-by: Catalin Marinas <catalin.marinas@arm.com>
Signed-off-by: Will Deacon <will@kernel.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 arch/arm64/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index dfa6a5cb99e4..12229cc5d908 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -53,7 +53,11 @@ $(warning Detected assembler with broken .inst; disassembly will be unreliable)
   endif
 endif
 
+ifeq ($(CONFIG_CC_IS_CLANG), y)
+COMPATCC ?= $(CC) --target=$(notdir $(CROSS_COMPILE_COMPAT:%-=%))
+else
 COMPATCC ?= $(CROSS_COMPILE_COMPAT)gcc
+endif
 export COMPATCC
 
 ifeq ($(CONFIG_GENERIC_COMPAT_VDSO), y)
-- 
2.24.0

